tools/src/main/scala/scala/scalanative/compiler/Compiler.scala

```diff
@@ -13,23 +13,26 @@ final class Compiler(opts: Opts) {
     Global.Member(Global.Top(opts.entry), "main_class.ssnr.ObjectArray_unit")
 
   private lazy val passCompanions: Seq[PassCompanion] = Seq(
-      pass.LocalBoxingElimination,
-      pass.DeadCodeElimination,
-      pass.MainInjection,
-      pass.ExternHoisting,
-      pass.ModuleLowering,
-      pass.RuntimeTypeInfoInjection,
-      pass.AsLowering,
-      pass.TraitLowering,
-      pass.ClassLowering,
-      pass.StringLowering,
-      pass.ConstLowering,
-      pass.SizeofLowering,
-      pass.UnitLowering,
-      pass.NothingLowering,
-      pass.ExceptionLowering,
-      pass.StackallocHoisting,
-      pass.CopyPropagation)
+      Seq(pass.LocalBoxingElimination, pass.DeadCodeElimination),
+      maybeInjectMain,
+      Seq(pass.MainInjection,
```

Comment:
You forgot to remove `MainInjection` from the list of passes. It's run twice now which causes the build to fail.